flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryPartition.java

```diff
@@ -33,6 +41,34 @@ public class BigQueryPartition {
 
     private BigQueryPartition() {}
 
+    /** Represents the partition types the BigQuery can use in partitioned tables. */
+    public enum PartitionType {
+        HOUR,
+        DAY,
+        MONTH,
+        YEAR,
+        INT_RANGE
+    }
+
+    public static StandardSQLTypeName retrievePartitionColumnType(
```

Comment:
Please add unit tests for this.

Reply:
Added improved implementation and test coverage across this file and `SchemaTransform`